src/http_workers/work_runner.rs

```diff
@@ -0,0 +1,147 @@
+use std::{sync::Arc, time::SystemTime};
+
+use async_trait::async_trait;
+use chrono::Utc;
+use hyper::{
+    client::{Builder, HttpConnector},
+    Body, Client, Method, Request,
+};
+
+use crate::{
+    cache::Cache,
+    identity::Identity,
+    twin::{SubstrateTwinDB, Twin, TwinDB},
+    types::{Message, QueuedMessage},
+    workers::Work,
+};
+
+use anyhow::{Context, Result};
+
+#[derive(Clone)]
+pub struct WorkRunner<C, I>
+where
+    C: Cache<Twin>,
+    I: Identity,
+{
+    twin_db: SubstrateTwinDB<C>,
+    identity: I,
+}
+
+impl<C, I> WorkRunner<C, I>
+where
+    C: Cache<Twin>,
+    I: Identity,
+{
+    pub fn new(twin_db: SubstrateTwinDB<C>, identity: I) -> Self {
+        Self { twin_db, identity }
+    }
+
+    async fn get_twin(&self, twin_id: usize, retires: usize) -> Option<Twin> {
+        for _ in 0..retires {
+            let twin = self.twin_db.get(twin_id.to_owned() as u32).await;
+
+            match twin {
+                Ok(twin) => {
+                    return Some(twin);
+                }
+                Err(err) => {
+                    log::debug!(
+                        "can not retrieve twin from substrate for this id '{}' because of {}",
+                        twin_id,
+                        err
+                    );
+                    continue;
+                }
+            };
+        }
+        None
+    }
+
+    fn encrypt_dat(dat: String, pubkey: String) -> Result<String> {
+        todo!()
+    }
+
+    async fn send_msg(twin: Twin, uri_path: String, msg: Message, dst: usize) -> Result<()> {
+        let req = Request::builder()
+            .method(Method::POST)
+            .header("content-type", "application/json");
+
+        let req = req.uri(format!(
```

Comment:
using `format!` to build a valid URL is wrong. already the Twin.address is just an IP (v6). so this will produce something like `<ip>/rmb-remote` which is not a valid URL. a valid URL is instead should look like
```
http://<ip>[:port]/rmb-remote`
```
The port for RMB is fixed to `8051` always.

Anyway there are so many edge cases for building a URL from it's parts (including url encoding). It's safer to use a URL specific crate. 

For example lets check on those following case
- address = `192.168.1.10` (ipv4)
- address = `202:6df5:9559:4c41:fa57:b09f:6e:ee0f` (ipv6)
- address = "some-name.com" (host-name)
- Also should we support allowing twins to include a custom port in the address in case they going to run their twin on a different port than `8051`

So for each case the url will look a little bit different like
- url = `http://192.168.1.10:8051/rmb-remote`
- url = `http://[202:6df5:9559:4c41:fa57:b09f:6e:ee0f]:8051/rmb-remote`
- url = `http://some-name.com:8051/rmb-remote`

If we gonna allow Twin objects to include ports in their own addresses. So let's say the Twin address is already set to `[202:6df5:9559:4c41:fa57:b09f:6e:ee0f]:9000` so the result url should then equal to `http://[202:6df5:9559:4c41:fa57:b09f:6e:ee0f]:9000`